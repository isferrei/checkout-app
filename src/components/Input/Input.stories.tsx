import type { Meta, StoryObj } from "@storybook/react";

import type { InputProps } from "./index";
import { Input } from "./index";

// More on how to set up stories at: https://storybook.js.org/docs/7.0/react/writing-stories/introduction
const meta = {
  title: "Example/Input",
  component: Input,
  tags: ["autodocs"],
  parameters: {
    // More on how to position stories at: https://storybook.js.org/docs/7.0/react/configure/story-layout
    layout: "fullscreen",
  },
} satisfies Meta<InputProps>;

export default meta;
type Story = StoryObj<InputProps>;

export const Default = {
  args: {
    placeholder: "DD/MM/AAAA",
    onChange: () => null,
    mask: "99/99/9999",
    type: "text",
  },
} satisfies Story;
