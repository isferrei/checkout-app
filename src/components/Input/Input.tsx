import InputMask from "react-input-mask";

export type InputProps = {
  placeholder: string;
  onChange: (event: React.FormEvent<HTMLFormElement>) => void;
  mask: string;
  type?: string;
  defaultValue?: string;
  label?: string;
  value?: string;
  name: string;
};

export function Input(props: InputProps) {
  return (
    <div>
      <label className="text-darkGray text-[0.8rem]" htmlFor={props.name}>
        {props.label || ""}
      </label>
      <InputMask
        mask={props.mask}
        className="w-full border-b-[1px] text-[1rem] text-darkGray font-sans border-[#F4F3F6] py-[0.25rem] pl-1 placeholder:text-lightGray focus:outline-none focus:ring-2"
        type={props.type ?? "text"}
        placeholder={props.placeholder}
        onChange={() => props.onChange}
        defaultValue={props.defaultValue ?? ""}
        value={props.value}
        name={props.name}
      />
    </div>
  );
}
