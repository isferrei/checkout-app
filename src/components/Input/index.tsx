import InputMask from "react-input-mask";

type InputProps = {
  placeholder: string;
  onChange: (event: React.FormEvent<HTMLFormElement>) => void;
  mask: string;
  type?: string;
  defaultValue?: string;
};

export function Input(props: InputProps) {
  const handleCreateNewCheckout = () => {};

  return (
    <InputMask
      mask={props.mask}
      className="w-full rounded-full border border-blackBlue py-2 pl-4 pr-12 drop-shadow-[0_5px_10px_#101E3224] focus:outline-none focus:ring-2"
      type={props.type ?? "text"}
      placeholder={props.placeholder}
      onChange={() => props.onChange}
      defaultValue={props.defaultValue ?? ""}
    />
  );
}
