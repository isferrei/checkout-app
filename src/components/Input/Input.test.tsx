import React from "react"
import { render, screen } from "@testing-library/react"
import type { ReactNode } from "react"

import { Input } from "./Input"

jest.mock(
    "next/head",
    () =>
        function Head(props: { children: ReactNode }) {
            return props.children
        }
)

describe("Input component", () => {
    describe("Render method", () => {
        it("should match the snapshot", () => {
            const placeholder = "DD/MM/AAAA"
            const mask = "99/99/9999"
            const type = "text"
            const onChange = () => false

            const { container } = render(
                <Input
                    placeholder={placeholder}
                    mask={mask}
                    type={type}
                    onChange={onChange}
                    name={""}
                />
            )

            expect(container).toMatchSnapshot()
        })
    })
})
