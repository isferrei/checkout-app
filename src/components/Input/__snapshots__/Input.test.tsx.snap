// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Render method should match the snapshot 1`] = `
<div>
  <div>
    <label
      class="text-darkGray text-[0.8rem]"
      for=""
    />
    <input
      class="w-full border-b-[1px] text-[1rem] text-darkGray font-sans border-[#F4F3F6] py-[0.25rem] pl-1 placeholder:text-lightGray focus:outline-none focus:ring-2"
      id=""
      name=""
      placeholder="DD/MM/AAAA"
      type="text"
      value=""
    />
  </div>
</div>
`;
